src/LodeRunner.Core/Models/Validators/TestRunValidator.cs

```diff
@@ -17,13 +17,35 @@ public class TestRunValidator : BaseEntityValidator<TestRun>
         /// </summary>
         public TestRunValidator()
         {
+            var minDate = new DateTime(1990, 1, 1);
+
+            // Validations for BaseEntityModel Fields
             this.RuleFor(m => m.Id).NotEmpty();
+            this.RuleFor(m => m.Id)
+                .Must(id => Guid.TryParse(id, out _))
+                .WithMessage("Id must be GUID.");
             this.RuleFor(m => m.PartitionKey).NotEmpty();
             this.RuleFor(m => m.EntityType).IsInEnum();
-            this.RuleFor(m => m.CreatedTime).GreaterThan(DateTime.MinValue).WithMessage("Creation time is required.");
+            this.RuleFor(m => m.Name).NotEmpty();
+
+            // Validation For Non-Entity Properties in TestRun
+            this.RuleFor(m => m.CreatedTime).GreaterThan(minDate);
+            this.RuleFor(m => new { m.StartTime, m.CreatedTime} )
+                .Must(m => m.StartTime >= m.CreatedTime )
+                .WithMessage("Start Time Must Be Greater Than Or Equal To Created Time");
+            this.RuleFor(m => new { m.CompletedTime, m.StartTime})
+                .Must(m => m.CompletedTime >= m.StartTime )
+                .When(m => m.CompletedTime != null)
+                .WithMessage("Completed Time Must Be Greater Than Or Equal To Start Time");
+
+            // Validation For Entity Properties in TestRun
+            this.RuleFor(m => m.LoadTestConfig)
+                .NotEmpty();
             this.RuleFor(m => m.LoadTestConfig).SetValidator(new LoadTestConfigValidator());
-            this.RuleFor(m => m.LoadClients.Count).GreaterThan(0);
+            this.RuleFor(m => m.LoadClients)
+                .NotEmpty();
             this.RuleForEach(m => m.LoadClients).SetValidator(new LoadClientValidator());
```

Comment:
This does not have to be answered with this PR - but curious if it is valid/possible for the same LoadClient to be listed more than once for a TestRun.